src/main/java/com/selfxdsd/selfpm/PayInvoices.java

```diff
@@ -0,0 +1,102 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+ * OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.selfxdsd.selfpm;
+
+import com.selfxdsd.api.Contract;
+import com.selfxdsd.api.Invoice;
+import com.selfxdsd.api.Project;
+import com.selfxdsd.api.ProjectManager;
+import com.selfxdsd.api.Self;
+import com.selfxdsd.api.Wallet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+/**
+ * Each PM will periodically verify their Project Contract Invoices an try to
+ * pay the ones that are eligible.
+ * @author criske
+ * @version $Id$
+ * @since 0.0.6
+ * @checkstyle NestedForDepth (500 lines)
+ */
+@Component
+public final class PayInvoices {
+
+    /**
+     * The PMs will verify unpaid invoices and try to pay them every MONDAY.
+     */
+    private static final String EVERY_MONDAY = "0 0 * * MON";
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(
+        PayInvoices.class
+    );
+
+    /**
+     * Self's core.
+     */
+    private final Self selfCore;
+
+    /**
+     * Ctor.
+     * @param selfCore Self Core, injected by Spring automatically.
+     */
+    @Autowired
+    public PayInvoices(final Self selfCore) {
+        this.selfCore = selfCore;
+    }
+
+    /**
+     * Every Monday the PMs should verify their Project Contract Invoices an
+     * try to pay the ones that are eligible.
+     */
+    @Scheduled(cron = EVERY_MONDAY)
+    public void payInvoices() {
+        LOG.debug("Checking invoices to be paid");
+        for(final ProjectManager manager : this.selfCore.projectManagers()) {
+            for(final Project project : manager.projects()) {
+                final Wallet wallet = project.wallet();
+                for(final Contract contact: project.contracts()){
+                    for(final Invoice invoice: contact.invoices()){
+                        if (!invoice.isPaid()) {
+                            LOG.debug(
+                                manager.username()
+                                    + " is trying to pay invoice #"
+                                    + invoice.invoiceId()
+                                    + " for contract: "
+                                    + contact.contractId()
+                            );
+                            wallet.pay(invoice);
```

Comment:
@criske We should also verify if the Invoice is over 108 EUR beforehand. It will work like this as well, but the ``Payment Failed`` error will appear on Self Web and we don't want that. 

Also, the whole payment should be in a ``try/catch (Exception)``, so we can catch any missed exceptions (if we don't catch them, the payment process will stop at the first problem and the subsequent Projects will not be iterated).

Reply:
@criske After wallet.pay(...)  you can say ``break`` (a Contract never has more than 1 active Invoice).

Reply:
> @criske We should also verify if the Invoice is over 108 EUR beforehand. It will work like this as well, but the `Payment Failed` error will appear on Self Web and we don't want that.
> 

Since this is a scheduler, shouldn't this work separately from `self-web` or...?. Also I think the wallet is already decorated from `self-storage`, so the 108EUR check/exception should be already handled. 
But yeah, a try/catch wouldn't hurt just to be safe.

Reply:
@criske 

> Since this is a scheduler, shouldn't this work separately from self-web or...?

Yes, it works separately from self-web. Maybe I don't understand the question.

> Also I think the wallet is already decorated from self-storage, so the 108EUR check/exception should be already handled.

Yes, it is handled, but if the Invoice is under 108, a failed Payment will be registered and we do not want that. So basically the decorator is just a precaution, in case the check hasn't been done earlier. The point is, we should not even try to pay the Invoice if it's under 108 :D

Reply:
@amihaiemil 

> @criske
> Yes, it works separately from self-web. Maybe I don't understand the question.
> 
You said _"..but the Payment Failed error will appear on Self Web and we don't want that."_ , but now that you explained that we don't want failed Payment to be registered from here, I think I understand how could this "leak" to `self-web`

Any reason for why we don't want to register the failed payments here? :D

Reply:
@criske Yes, the failed Payment will appear in self-web because it will be saved in the DB. Remember, self-pm uses the same core and storage as self-web :D 

We want to show failed payments as rarely as possible because the users lose trust when they see failed payments, usually. So wherever possible, we should avoid failed payments, I think.

This 108 EUR failed payment case should never occur (because we have control and can do this check). Ideally, failed payment for 108 EUR case will only occur if the user hacks the front-end and bypasses the validation.